extra_data/tests/test_filecache.py

```diff
@@ -32,17 +26,29 @@ def test_filecache_large(mock_spb_raw_run, filecache_512):
     files = [os.path.join(mock_spb_raw_run, f) \
              for f in os.listdir(mock_spb_raw_run) if f.endswith('.h5')]
     run = DataCollection.from_paths(files)
+    
+    # estimate the number of files which should be open to read the first train
+    cnt, fix = {}, {}
+    for f in run.files:
+        a, b = os.path.basename(f.filename).rsplit('-',1)
+        try:
+            cnt[a] += 1
+        except KeyError:
+            cnt[a] = 1
+        if b == 'S00000.h5':
+            fix[a] = cnt[a] 
+    nfiles = sum(a for a in fix.values())
```

Comment:
I think we should probably just drop the `len(fc._cache)` check instead of adding all this logic, or make it more vague (e.g. `>= 17`). Going to all this trouble to figure out what the number should be seems like it would tie the test to the implementation details.

Reply:
Yes it would. But this is also a reminder of the specifics of the implementation. I spent half a day to figure this out.
As an option, we can sort the files and make the implementation deterministic. In `DataCollection.from_paths()` replace:
```python
    return cls(files, ctx_closes=True)
```
with
```python
    return cls(sorted(files, key=lambda f: f.filename), ctx_closes=True)
```
I prefer the second option. What do you think?